src/ClubInfo/Links.tsx

```diff
@@ -10,26 +12,59 @@ const LinkListDiv = styled.ul`
     margin-left: 0;
     margin-bottom: 10px;
   }
+  overflow: auto;
   margin-top: 20px;
   margin-left: 15%;
 `;
 
+const urlStart = /^https?:\/\/(www\.)?/;
+const urlQueries = /\?.*$/;
+
+function simplifyURL(url: string) {
+  return url.replace(urlStart, '').replace(urlQueries, '');
+}
+
+function getIcon(url: string): IconType {
+  const defaultIcon: IconType = FaLink;
+  const urlMap: { [key: string]: IconType } = {
+    'facebook.com': FaFacebook,
+    'instagram.com': FaInstagram,
+    'discord.gg': FaDiscord,
+  };
+
+  const domain = new URL(url).hostname.replace(/^www\./, '');
+
+  return urlMap[domain] ?? defaultIcon;
+}
+
 function LinkItem(props: { link: string }) {
   return (
-    <div>
-      <Icon icon={IoMdLink}></Icon>
-      <p>{props.link}</p>
-    </div>
+    <Button size="sm" style={{ maxWidth: '100%' }}>
+      <a
+        href={props.link}
+        style={{
+          overflow: 'hidden',
+          textOverflow: 'ellipsis',
+          display: 'flex',
+          alignItems: 'center',
+        }}
+      >
+        <Icon icon={getIcon(props.link)}></Icon>
+        <span style={{ textOverflow: 'ellipsis', overflow: 'hidden' }}>{simplifyURL(props.link)}</span>
+      </a>
```

Comment:
It's currently opening links in new tabs and closing the directory website which isn't ideal

Reply:
this should be fixed in bad0773